File: examples/api/nodejs/create_client.js
#!/usr/bin/env node
/**
 * Create a new WireGuard client
 */

const axios = require('axios');

// Get API configuration from environment
const API_KEY = process.env.WIREGUARD_API_KEY;
const BASE_URL = process.env.WIREGUARD_BASE_URL;

if (!API_KEY) {
  console.error('Error: WIREGUARD_API_KEY environment variable not set');
  process.exit(1);
}

if (!BASE_URL) {
  console.error('Error: WIREGUARD_BASE_URL environment variable not set');
  process.exit(1);
}

// Create API client
const api = axios.create({
  baseURL: BASE_URL,
  headers: {
    'Authorization': `Bearer ${API_KEY}`,
    'Content-Type': 'application/json'
  },
  timeout: 30000
});

async function createClient(name, email, group, ipRange = '10.8.0.0/24') {
  try {
    const response = await api.post('/api/v1/client', {
      name,
      email,
      group,
      allocated_ips: [ipRange],
      allowed_ips: ['0.0.0.0/0'],
      use_server_dns: true,
      enabled: true
    });
    
    const client = response.data;
    console.log(`✓ Client '${name}' created successfully!`);
    console.log(`  ID: ${client.id}`);
    console.log(`  Public Key: ${client.public_key}`);
    console.log(`  Allocated IPs: ${client.allocated_ips.join(', ')}`);
    
    return client;
    
  } catch (error) {
    if (error.response) {
      console.error('✗ Error creating client:', error.response.data.message || error.response.data);
    } else {
      console.error('✗ Connection error:', error.message);
    }
    process.exit(1);
  }
}

// Parse command line arguments
const args = process.argv.slice(2);
if (args.length < 3) {
  console.log('Usage: node create_client.js <name> <email> <group> [ip_range]');
  console.log('Example: node create_client.js "John Doe" "john@example.com" "Employees" "10.8.0.50/32"');
  process.exit(1);
}

const [name, email, group, ipRange] = args;

// Run
createClient(name, email, group, ipRange);

File: examples/api/python/create_client.py
#!/usr/bin/env python3
"""
Create a new WireGuard client
"""

import os
import sys
import requests
import argparse

def get_api_config() -> tuple:
    """Get API configuration from environment variables"""
    api_key = os.getenv('WIREGUARD_API_KEY')
    base_url = os.getenv('WIREGUARD_BASE_URL')
    
    if not api_key:
        print("Error: WIREGUARD_API_KEY environment variable not set", file=sys.stderr)
        sys.exit(1)
    
    if not base_url:
        print("Error: WIREGUARD_BASE_URL environment variable not set", file=sys.stderr)
        sys.exit(1)
    
    return api_key, base_url

def create_client(name: str, email: str, group: str, ip_range: str = "10.8.0.0/24"):
    """Create a new WireGuard client"""
    api_key, base_url = get_api_config()
    
    headers = {
        'Authorization': f'Bearer {api_key}',
        'Content-Type': 'application/json'
    }
    
    payload = {
        'name': name,
        'email': email,
        'group': group,
        'allocated_ips': [ip_range],
        'allowed_ips': ['0.0.0.0/0'],
        'use_server_dns': True,
        'enabled': True
    }
    
    try:
        response = requests.post(
            f'{base_url}/api/v1/client',
            headers=headers,
            json=payload,
            timeout=30
        )
        
        if response.status_code == 200:
            client = response.json()
            print(f"✓ Client '{name}' created successfully!")
            print(f"  ID: {client['id']}")
            print(f"  Public Key: {client['public_key']}")
            print(f"  Allocated IPs: {', '.join(client['allocated_ips'])}")
            return client
        else:
            error = response.json()
            print(f"✗ Error creating client: {error.get('message', 'Unknown error')}", file=sys.stderr)
            sys.exit(1)
            
    except requests.exceptions.RequestException as e:
        print(f"✗ Connection error: {e}", file=sys.stderr)
        sys.exit(1)

def main():
    parser = argparse.ArgumentParser(description='Create a new WireGuard client')
    parser.add_argument('name', help='Client name')
    parser.add_argument('email', help='Client email')
    parser.add_argument('group', help='Client group')
    parser.add_argument('--ip', default='10.8.0.0/24', help='IP allocation (default: 10.8.0.0/24)')
    
    args = parser.parse_args()
    
    create_client(args.name, args.email, args.group, args.ip)

if __name__ == '__main__':
    main()
